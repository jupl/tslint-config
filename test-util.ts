import {readFile} from 'mz/fs'
import {
  Configuration,
  FormatterConstructor,
  IRuleFailureJson,
  Linter,
  RuleFailure,
} from 'tslint'

export interface ILintResult {
  readonly failures: ReadonlyArray<IRuleFailureJson>
  readonly errorCount: number
  readonly warningCount: number
  readonly fixes: ReadonlyArray<IRuleFailureJson>
  readonly format: string | FormatterConstructor
  readonly output: string
}

const basePath = process.cwd()
const configuration = Configuration.loadConfigurationFromPath('./tslint.json')
const linter = new Linter({fix: true}, Linter.createProgram('./tsconfig.json'))

export async function lintFixture(fixtureName: string): Promise<ILintResult> {
  const fileName = `./fixtures/${fixtureName}`
  const file = await readFile(fileName, 'utf8')
  linter.lint(fileName, file, configuration)
  const {failures, output, fixes = [], ...result} = linter.getResult()
  return {
    ...result,
    failures: failures.map(failureToJson),
    fixes: fixes.map(failureToJson),
    output: output.split(basePath).join('.'),
  }
}

function failureToJson(failure: RuleFailure): IRuleFailureJson {
  const {name, ...json} = failure.toJson()
  return {...json, name: name.split(basePath).join('.')}
}
