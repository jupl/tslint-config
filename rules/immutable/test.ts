import {lintFixture} from '../../test-util'

test('Immutable rules should be valid', async() => {
  const results = await lintFixture('immutable.ts')
  expect(results).toMatchSnapshot()
})
