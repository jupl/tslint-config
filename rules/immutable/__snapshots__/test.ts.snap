// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Immutable rules should be valid 1`] = `
Object {
  "errorCount": 5,
  "failures": Array [
    Object {
      "endPosition": Object {
        "character": 8,
        "line": 6,
        "position": 74,
      },
      "failure": "Modifying properties of existing object not allowed.",
      "fix": Array [],
      "name": "./fixtures/immutable.ts",
      "ruleName": "no-object-mutation",
      "ruleSeverity": "ERROR",
      "startPosition": Object {
        "character": 0,
        "line": 6,
        "position": 66,
      },
    },
    Object {
      "endPosition": Object {
        "character": 11,
        "line": 7,
        "position": 86,
      },
      "failure": "Modifying properties of existing object not allowed.",
      "fix": Array [],
      "name": "./fixtures/immutable.ts",
      "ruleName": "no-object-mutation",
      "ruleSeverity": "ERROR",
      "startPosition": Object {
        "character": 0,
        "line": 7,
        "position": 75,
      },
    },
    Object {
      "endPosition": Object {
        "character": 8,
        "line": 6,
        "position": 74,
      },
      "failure": "Mutating an array is not allowed.",
      "fix": Array [],
      "name": "./fixtures/immutable.ts",
      "ruleName": "no-array-mutation",
      "ruleSeverity": "ERROR",
      "startPosition": Object {
        "character": 0,
        "line": 6,
        "position": 66,
      },
    },
    Object {
      "endPosition": Object {
        "character": 11,
        "line": 7,
        "position": 86,
      },
      "failure": "Mutating an array is not allowed.",
      "fix": Array [],
      "name": "./fixtures/immutable.ts",
      "ruleName": "no-array-mutation",
      "ruleSeverity": "ERROR",
      "startPosition": Object {
        "character": 0,
        "line": 7,
        "position": 75,
      },
    },
    Object {
      "endPosition": Object {
        "character": 9,
        "line": 8,
        "position": 96,
      },
      "failure": "Mutating an array is not allowed.",
      "fix": Array [],
      "name": "./fixtures/immutable.ts",
      "ruleName": "no-array-mutation",
      "ruleSeverity": "ERROR",
      "startPosition": Object {
        "character": 0,
        "line": 8,
        "position": 87,
      },
    },
  ],
  "fixes": Array [],
  "format": "prose",
  "output": "
ERROR: ./fixtures/immutable.ts[7, 1]: Modifying properties of existing object not allowed.
ERROR: ./fixtures/immutable.ts[7, 1]: Mutating an array is not allowed.
ERROR: ./fixtures/immutable.ts[8, 1]: Modifying properties of existing object not allowed.
ERROR: ./fixtures/immutable.ts[8, 1]: Mutating an array is not allowed.
ERROR: ./fixtures/immutable.ts[9, 1]: Mutating an array is not allowed.
",
  "warningCount": 0,
}
`;
